presto-main/src/main/java/com/facebook/presto/operator/Operator.java

```diff
@@ -73,6 +73,13 @@
     Page getOutput();
 
     /**
+     * After calling this method operator should revoke all reserved revocable memory
+     */
+    default void revokeMemory()
```

Comment:
shouldn't that return future too? Why block driver if operators might offload spill to background IO threads.